crates/ruff_server/src/session/settings.rs

```diff
@@ -130,7 +130,8 @@ enum InitializationOptions {
         workspace_settings: Vec<WorkspaceSettings>,
     },
     GlobalOnly {
-        settings: Option<ClientSettings>,
+        #[serde(flatten)]
+        settings: ClientSettings,
```

Comment:
@snowsignal when you're back, can you explain why is this being flattened? This is causing #11507 because now it's not under `settings` but rather in the global table.